types/gatherer.d.ts

```diff
@@ -54,11 +57,14 @@ declare global {
       driver: Driver;
       passConfig: Config.Pass
       settings: Config.Settings;
+      computedCache: Map<string, ArbitraryEqualityMap>
       /** Gatherers can push to this array to add top-level warnings to the LHR. */
       LighthouseRunWarnings: Array<string | IcuMessage>;
       baseArtifacts: BaseArtifacts;
     }
 
+    export type CompatibilityContext = PassContext | FRTransitionalContext;
```

Comment:
`FRTransitionalContext` was supposed to be the minimal set of shared context ("The limited context interface shared between pre and post Fraggle Rock Lighthouse."). 

I get that `dependencies` somewhat violated that by always being `{}`, and I'm guessing that's what prompted this addition? I like the explicitness. Perhaps we should enforce that at the shared FRGatherer implementation level? Like if the `meta` contains dependencies throw an error in `afterPass` that it must be overriden to handle dependencies?

Reply:
How do you feel about these changes adding `dependencies: {}` to `PassContext`?

Reply:
I'm not sure that solves the issue I have in mind, but maybe it's aligned and just doesn't affect things, so I'm open to it. 

I'm thinking all `afterPass` methods should be typed `PassContext` and all `getArtifact` should be typed `FRTransitionalContext`. In `base-gatherer`'s `afterPass` method at runtime we check if there are any dependencies defined in the meta, if there are we throw an error immediately and don't invoke the other methods.

From that lens it doesn't really matter if `dependencies` is defined on `PassContext` because nothing is going to attempt to access dependencies on something that is typed `PassContext` anyway.

Reply:
Ah I think I see now.